src/ai/defaultai.cc

```diff
@@ -3350,6 +3365,61 @@ void DefaultAI::diplomacy_actions(const Time& gametime) {
                                           (accept ? Widelands::DiplomacyAction::kAcceptJoin :
                                                     Widelands::DiplomacyAction::kRefuseJoin)),
 			                             pda.sender);
+			verb_log_dbg_time(
+			   gametime,
+			   "AI Diplomacy: Player(%d), %s the invitation of player (%d) with diploscore: %d\n",
+			   static_cast<unsigned int>(pda.other), accept ? "accepts" : "denies",
+			   static_cast<unsigned int>(pda.sender), player_statistics.get_diplo_score(pda.sender));
+			if (accept) {
+				request_accepted = true;
+			}
+		}
+	}
+	if (request_accepted || me_def) {
+		return;
+	}
+
+	for (Widelands::PlayerNumber opn = 1; opn <= game().map().get_nrplayers(); ++opn) {
+		const Widelands::Player* other_player = game().get_player(opn);
+		// other player needs to exist and different from us
+		// we need to be still alive and we don't have send a request in last 5 + X minutes
+		if (other_player == nullptr || opn == mypn ||
+		    player_statistics.player_diplo_requested_lately(opn, gametime)) {
+			continue;
+		}
+		if (player_statistics.get_diplo_score(opn) >= 40) {
```

Comment:
```suggestion
		if (player_statistics.get_diplo_score(opn) >= 40) &&
		    me->team_number() != other_player->team_number()) {
```

Reply:
this change would prevent 2 players in team 0 tom team. Did you see a lot of regressions of players inviting each other. 
I intended to prevent this by requesting different team powers for the invitation. If this happens we need some logging because that means team powers are calculated differently.

Reply:
ok I will ad some logging and run a test,

Reply:
BTW instead of military power only, we should sum or average the diploscores of unaffected members of both teams, and decide which team is better comparing that.

Reply:
BTW instead of military power only, we should sum or average the diploscores of the unaffected members of both teams, and compare that. Or even if both are negative, then leave current team and send invite in the same step.

Reply:
We could do that but I think this might be a lot of effort just to distinguish between invite and join.

Reply:
I think (based on the test games too) that it's a very important distinction. :)

Reply:
ok so I'll exchange military team power with team diploscores correct?

Reply:
Leaving out the player making the request and the player who is requested, because they will be both in the selected team if all works out. So they have to evaluate the other members of both teams to see which ones they're better off with. I think I'll write an invite or join function proposal, because I have several things in mind how to handle various cases.

Reply:
Basically I thought about just summing the diplo_scores in recalculate team power as it is least costly.